### PR TITLE
Adding healthcheck info and lock into gw struct

### PR DESCRIPTION
**Pull request by tbuchaillot**
## Description

Adding Healthcheck info and lock into gateway struct to add consistency and avoid overlapping values in MDCB testing framework

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [ ] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
  - [ ] If new config option added, ensure that it can be set via ENV variable
- [ ] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] When updating library version must provide reason/explanation for this update.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Check your code additions will not fail linting checks:
  - [ ] `go fmt -s`
  - [ ] `go vet`

### CODE REVIEW

File: gateway/server.go [hunk 1]
```diff
@@ -179,6 +180,9 @@ type Gateway struct {
 	// RedisController keeps track of redis connection and singleton
 	RedisController *storage.RedisController
 	hostDetails     hostDetails
+
+	healthCheckInfo atomic.Value
+	healthCheckLock sync.Mutex
```
Comment (furkansenharputlu) [inline gateway/server.go:185]: if `healthCheckInfo` is atomic, why do we need a mutex to guard it?

File: gateway/server.go [hunk 2]
```diff
@@ -82,6 +82,7 @@ var (
 const appName = "tyk-gateway"
 
 type Gateway struct {
+
```
Comment (furkansenharputlu) [inline gateway/server.go:85]: I think it is a leftover

File: gateway/server.go (hunk 1 above)
Comment (tbuchaillot) [reply gateway/server.go]: good point @furkansenharputlu , probably old code. Removed the mutex 👍

**Review by furkansenharputlu — APPROVED** [review]

**Review by matiasinsaurralde — APPROVED** [review]

**Review by jeffy-mathew — APPROVED** [review]

**Pull request merged**